src/core/modules/impl/DefaultCreateContract.ts

```diff
@@ -16,10 +16,13 @@ export class DefaultCreateContract implements CreateContract {
 
     const srcTx = await this.arweave.createTransaction({ data: src }, wallet);
 
+    const contractType: ContractType = src instanceof Buffer ? 'wasm' : 'js';
+
     srcTx.addTag(SmartWeaveTags.APP_NAME, 'SmartWeaveContractSource');
     // TODO: version should be taken from the current package.json version.
     srcTx.addTag(SmartWeaveTags.APP_VERSION, '0.3.0');
-    srcTx.addTag('Content-Type', 'application/javascript');
+    srcTx.addTag(SmartWeaveTags.SDK, 'RedStone');
+    srcTx.addTag(SmartWeaveTags.CONTENT_TYPE, contractType == 'js' ? 'application/javascript' : 'application/wasm');
```

Comment:
contracts indexing in redstone-sw-gateway has to adopted to this..as the gateway keeps all contracts source in 'text' column 🤦

Reply:
We don't need to implement it now, but
1) It might be easier for devs to work on the source code and pass raw source code to the sdk abstracting the compilation step
2) It may also be easier for debugging, monitoring to have access to the original contract's source code, at least in one of the columns in our gateway